compiler/rustc_hir_analysis/src/check/method/suggest.rs

```diff
@@ -1201,6 +1204,80 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
         false
     }
 
+    fn suggest_range_for_iter(
+        &self,
+        tcx: TyCtxt<'tcx>,
+        actual: Ty<'tcx>,
+        source: SelfSource<'tcx>,
+        span: Span,
+        item_name: Ident,
+        ty_str: &str,
+    ) -> bool {
+        if let SelfSource::MethodCall(expr) = source {
+            for (_, parent) in tcx.hir().parent_iter(expr.hir_id).take(5) {
+                if let Node::Expr(parent_expr) = parent {
+                    let lang_item = match parent_expr.kind {
```

Comment:
would be nice to only have this match once. Maybe change `is_range_expr` to call `get_range_expr_ty` which returns `Option<LangItem>` while using `get_range_expr_ty` directly for this

Reply:
There is no `is_range_expr`, so I guess you mean `is_range_literal` ?
There is a trivial difference, here we can not handle the case for:
`RangeFrom` and `RangeFull`,
we only handle the scenarios which have a right boundry...